contents/blog/clickhouse-vs-postgres.md

```diff
@@ -0,0 +1,186 @@
+
+Imagine you’ve abandoned your dating app and banking businesses; now, you run a lovely education non-profit. You need to constantly query the median test score from your database — you could construct a Materialized View that stores median test scores by grade level. Then, when you need to access those test scores, you can query the Materialized View and access the results without re-calculating the median scores. 
+
+Here’s a version using our previous simple hedgehog database:
+
+![clickhouse resource use](../images/blog/clickhouse-vs-postgres/visualizing-materialized-views.png) 
+
+#### Materialized View averaging age
+
+Both Postgres *and* ClickHouse have functionality to build Materialized Views; the fundamental difference is that Postgres’s Materialized Views need to be manually re-updated, whereas ClickHouse automatically updates them. For some cases, like test scores, this can be done once, after all the tests are uploaded; for others, like website analytics, an auto-update materialized view like ClickHouse’s is necessary. 
+
+ClickHouse can only accomplish auto-updates efficiently because of its insert-and-optimize-later philosophy. ClickHouse never sleeps — it uses its idle time to compress data in Materialized Views so that future data look-ups are fast and efficient. 
+
+But it’s more than just time-allocation. For instance, if you need to add numbers to produce aggregate results, ClickHouse’s SummingMergeTree engine will dramatically parallelize the process. Alternatively, if you need to average data, the AggregatingMergeTree is your dear friend. 
+
+Ironically, the more interesting topic in Materialized Views is not how ClickHouse efficiently updates them — that is the bread and butter of its purpose — but if Postgres can be hacked to achieve something similar. 
+
+Here are some strategies on how Postgres can emulate automatic Materialized View updates alongside their downsides: 
+
+- **Fast Refresh Module.** By blending a static Materialized View and a running log of changes, Postgres’s Fast Refresh Module can simulate an automatically-updated Materialized View. The downsides are that utilizing a log is incompatible with certain fetch requirements (e.g. count), high-write databases could overwhelm the log, and that logs slow down a fetch query.
+ 
+- **Triggers.** You could set a Postgres trigger to refresh a Materialized View on page load. This solution will rarely work unless you have seldom write operations; refreshing the Materialized View completely wipes the original result and will likely bludgeon the CPU. 
+
+![hacking postgres](../images/blog/clickhouse-vs-postgres/hacking-postgres-meme.png) 
+
+In a nutshell, Postgres can be bandaged up to achieve some efficiency that ClickHouse boasts around Materialized Views, but fundamentally, ClickHouse treats Materialized Views as an out-of-the-box benefit with efficiency *and* simplicity as cornerstone value props. 
```

Comment:
you might want to mention the databases out there that have actually hacked Postgres into a pretty solid OLAP db engine
- CitusDB
- TimescaleDB
- AWS Redshift
- Greenplum

Reply:
Definitely will mention—that's a good point. James also mentioned a Clickhouse vs TimescaleDB or something to similar effect has a lot of potential.